astro_rcnn.py

```diff
@@ -159,27 +163,45 @@ def load_image_disk(self, image_id, normalize='zscore'):
         # z-score normalization times factor A
         # A should be large enough to capture details in variety of images
         # and small enough to stay within 16-bit integer limits
-        if normalize == 'zscore':
-            A = self.A
-            g = A*(g - np.mean(g))/np.std(g)
-            r = A*(r - np.mean(r))/np.std(r)
-            z = A*(z - np.mean(z))/np.std(z)
-            #r = lupton_data[:,:,0]
-            #g = lupton_data[:,:,1]
-            #b = lupton_data[:,:,2]
-            # convert format
-            image = np.zeros([info['height'], info['width'], 3], dtype=np.int16)
+        
+        image = np.zeros([info['height'], info['width'], 3], dtype=np.int16)
+
+        if store_raw:
             image[:,:,0] = z # red
             image[:,:,1] = r # green
             image[:,:,2] = g # blue
-            self.images[image_id] = image
-            return image
-        elif normalize == 'lupton':
-            image = make_lupton_rgb(z, r, g, Q = self.Q, stretch = self.stretch)
-            self.images[image_id] = image
-            return image
-
-            
+            self.raws[image_id] = image
+
+        I = (z+r+g)/3.0
+        stretch = self.stretch
+        Q = self.Q
+
+        if normalize=='lupton':
+            z = z*np.arcsinh(stretch*Q*(I))/(Q*I)
+            r = r*np.arcsinh(stretch*Q*(I))/(Q*I)
+            g = g*np.arcsinh(stretch*Q*(I))/(Q*I)
+        elif normalize=='zscore':
+            #TODO np.percentile for 90th percentile
+            I = I*np.mean([np.std(g),np.std(r),np.std(z)])
+            z = (z - np.mean(z))/I
+            r = (r - np.mean(r))/I
+            g = (g - np.mean(g))/I
+
+        max_RGB = np.percentile([z,r,g], 99.95)
+        # avoid saturation
+        #if max_RGB > 1:
```

Comment:
Please remove lines:
`#TODO np.percentile for 90th percentile`
`#if max_RGB > 1:`